src/main/scala/org/allenai/pnp/BsoTrainer.scala

```diff
@@ -179,28 +183,26 @@ class BsoTrainer(val epochs: Int, val beamSize: Int, val maxIters: Int,
   }
 }
 
-class BsoPnpQueue[A](size: Int, val stateCost: ExecutionScore,
-    val graph: CompGraph, val log: LogFunction) extends PnpSearchQueue[A] {
+class BsoPnpQueue[A](size: Int, val stateCost: ExecutionScore) extends PnpSearchQueue[A] {
 
   val queue = new KbestQueue(size, Array.empty[(SearchState[A], Double)])
   val correctQueue = new KbestQueue(size, Array.empty[SearchState[A]])
   val incorrectQueue = new KbestQueue(size, Array.empty[SearchState[A]])
   
   val EXECUTION_INCORRECT_VAR_NAME = "**bso_execution_incorrect**"
 
-  override def offer(value: Pnp[A], env: Env, logProb: Double, tag: Any,
-      choice: Any, myEnv: Env): Unit = {
+  override def offer(value: Pnp[A], env: Env, logProb: Double, context: PnpInferenceContext, tag: Any, choice: Any): Unit = {
     if (logProb > Double.NegativeInfinity) {
-      log.startTimer("bso/beam_search/search_step/eval_cost")
+      context.log.startTimer("bso/beam_search/search_step/eval_cost")
       val cost = stateCost(tag, choice, env)
-      log.stopTimer("bso/beam_search/search_step/eval_cost")
-
+      context.log.stopTimer("bso/beam_search/search_step/eval_cost")
       val nextEnv = if (cost != 0.0) {
         env.setVar(EXECUTION_INCORRECT_VAR_NAME, null)
       } else {
         env
       }
-      
+
+      // TODO(joelgrus) is this right?
```

Comment:
It is for the current implementation (which just uses the scores as a correct/incorrect signal). I have a fix in the ICCV PR that I'll send out soon.